spec/services/displayable_pii_formatter_spec.rb

```diff
@@ -0,0 +1,210 @@
+
+        expect(result[:email]).to eq('test1@example.com')
+        expect(result[:all_emails]).to match_array(['test1@example.com', 'test2@example.com'])
+        expect(result[:verified_at]).to eq(I18n.l(verified_at, format: :event_timestamp))
+        expect(result[:x509_subject]).to eq('foo')
+        expect(result[:x509_issuer]).to eq('bar')
+        expect(result[:full_name]).to eq('Testy Testerson')
+        expect(result[:social_security_number]).to eq('900-12-3456')
+        expect(result[:address]).to eq('123 main st Washington, DC 20405')
+        expect(result[:birthdate]).to eq('January 01, 1990')
+        expect(result[:phone]).to eq('+1 202-212-1000')
+      end
+    end
+
+    describe '[:verified_at]' do
+      context 'for a verified user' do
+        let(:verified_at) { 1.day.ago }
+
+        it 'returns the date the user was verified' do
+          formatted_date = I18n.l(verified_at, format: :event_timestamp)
+          expect(formatter.format[:verified_at]).to eq(formatted_date)
+        end
+      end
+
+      context 'for an unverified user' do
+        let(:verified_at) { nil }
+
+        it 'returns a message about the user not being verified' do
+          expect(formatter.format[:verified_at]).to eq(
+            I18n.t('help_text.requested_attributes.verified_at_blank'),
+          )
+        end
+      end
+    end
+
+    describe 'PIV/CAC attributes' do
+      context 'the user has a piv/cac configured' do
+        let(:x509_issuer) { 'foo' }
+        let(:x509_subject) { 'bar' }
+
+        it 'returns x509_issuer and x509_subject' do
+          expect(formatter.format[:x509_issuer]).to eq(x509_issuer)
+          expect(formatter.format[:x509_subject]).to eq(x509_subject)
+        end
+      end
+
+      context 'the user does not have a piv/cac configured' do
+        let(:x509_issuer) { nil }
+        let(:x509_subject) { nil }
+
+        it 'returns nil for x509_issuer and x509_subject' do
+          expect(formatter.format[:x509_issuer]).to eq(nil)
+          expect(formatter.format[:x509_subject]).to eq(nil)
+        end
+      end
+    end
+
+    describe '[:address]' do
+      context 'without a second address line' do
+        let(:address1) { '123 main st' }
+        let(:address2) { '' }
+        let(:city) { 'Washington' }
+        let(:state) { 'DC' }
+        let(:zipcode) { '20405' }
+
+        it 'returns a formatted address' do
+          expect(formatter.format[:address]).to eq('123 main st Washington, DC 20405')
+        end
+      end
+
+      context 'with a second address line' do
+        let(:address1) { '123 main st' }
+        let(:address2) { 'apt 123' }
+        let(:city) { 'Washington' }
+        let(:state) { 'DC' }
+        let(:zipcode) { '20405' }
+
+        it 'returns a formatted address' do
+          expect(formatter.format[:address]).to eq('123 main st apt 123 Washington, DC 20405')
+        end
+      end
+    end
+
+    context '[:birthdate]' do
```

Comment:
yeah I think this is a good reason to just have methods exposed instead of hash keys